upf/io/pddl_reader.py

```diff
@@ -14,24 +14,321 @@
+            act.add_increase_effect(*eff if timing is None else (timing, *eff))
+        elif op == 'decrease':
+            eff = (self._parse_exp(problem, act, {}, exp[1]),
+                   self._parse_exp(problem, act, {}, exp[2]), cond)
+            act.add_decrease_effect(*eff if timing is None else (timing, *eff))
+        else:
+            eff = (self._parse_exp(problem, act, {}, exp), self._em.TRUE(), cond)
+            act.add_effect(*eff if timing is None else (timing, *eff))
+
+    def _add_condition(self, problem, act, exp, vars=None):
+        op = exp[0]
+        if op == 'and':
+            for e in exp[1:]:
+                self._add_condition(problem, act, e, vars)
+        elif op == 'forall':
+            vars_string = ' '.join(exp[1])
+            vars_res = parameters.parseString(vars_string)
+            if vars is None:
+                vars = {}
+            for g in vars_res['params']:
+                t = self._tm.UserType(g[1] if len(g) > 1 else 'object')
+                for o in g[0]:
+                    vars[o] = upf.model.Variable(o, t)
+            self._add_condition(problem, act, exp[2], vars)
+        elif len(exp) == 3 and op == 'at' and exp[1] == 'start':
+            cond = self._parse_exp(problem, act, {} if vars is None else vars, exp[2])
+            if vars is not None:
+                cond = self._em.Forall(cond, *vars.values())
+            act.add_condition(upf.model.StartTiming(), cond)
+        elif len(exp) == 3 and op == 'at' and exp[1] == 'end':
+            cond = self._parse_exp(problem, act, {} if vars is None else vars, exp[2])
+            if vars is not None:
+                cond = self._em.Forall(cond, *vars.values())
+            act.add_condition(upf.model.EndTiming(), cond)
+        elif len(exp) == 3 and op == 'over' and exp[1] == 'all':
+            t_all = upf.model.ClosedInterval(upf.model.StartTiming(), upf.model.EndTiming())
+            cond = self._parse_exp(problem, act, {} if vars is None else vars, exp[2])
+            if vars is not None:
+                cond = self._em.Forall(cond, *vars.values())
+            act.add_durative_condition(t_all, cond)
+        else:
+            raise upf.exceptions.UPFProblemDefinitionError(f'Not able to handle: {exp}')
+
+    def _add_timed_effects(self, problem, act, eff):
+        op = eff[0]
+        if op == 'and':
+            for e in eff[1:]:
+                self._add_timed_effects(problem, act, e)
+        elif len(eff) == 3 and op == 'at' and eff[1] == 'start':
+            self._add_effect(problem, act, eff[2], timing=upf.model.StartTiming())
+        elif len(eff) == 3 and op == 'at' and eff[1] == 'end':
+            self._add_effect(problem, act, eff[2], timing=upf.model.EndTiming())
+        else:
+            raise upf.exceptions.UPFProblemDefinitionError(f'Not able to handle: {eff}')
+
+
+    def parse_problem(self, domain_filename: str, problem_filename: str) -> 'upf.model.Problem':
```

Comment:
An interesting feature could be to parse just the domain and get a partial `upf.model.Problem`: the use-case would be to have a fixed domain written in PDDL that gets instantiated with dynamic data. 

Would it be possible to make the `problem_filename` optional here?